web/modules/custom/hoeringsportal_audit_log/src/Routing/RoutingChangeListener.php

```diff
@@ -0,0 +1,167 @@
+    $view = [];
+    $edit = [];
+
+    $editRoute = 'entity.node.edit_form';
+    $viewRoute = 'entity.node.canonical';
+
+    if ($contentTypes) {
+      // Loop through content types to categorize enabled audit log
+      // pages (view, edit).
+      foreach ($contentTypes as $contentType => $pages) {
+        foreach ($pages as $page => $enabled) {
+          // Categorize, only if the page is enabled (1 represents a checked
+          // checkbox in the form)
+          if ($enabled === 1) {
+            switch ($page) {
+              case 'view':
+                $view[] = $contentType;
+                break;
+
+              case 'edit':
+                $edit[] = $contentType;
+                break;
+            }
+          }
+        }
+      }
+    }
+    // Early return if there is not assigned audit log to any pages.
+    if (empty($view) && empty($edit) && empty($loggedRouteNames)) {
+      return;
+    }
+
+    // Check if the route name is in the array of hardcoded routes
+    // from the settings file.
+    if ($loggedRouteNames && in_array($routeName, $loggedRouteNames)) {
+      $this->logAuditMessage($pathInfo);
+      return;
+    }
+
+    // Get node to get content type of current page.
+    $nodeId = $this->routeMatch->getRawParameter('node');
+    $node = NULL;
+
+    if ($nodeId) {
+      $node = Node::load($nodeId);
+
+      $routes = [
+        [$edit, $editRoute],
+        [$view, $viewRoute],
+      ];
+      foreach ($routes as [$page, $route]) {
+        if ($this->auditOnNodePage($routeName, $page, $route, $node)) {
+          // If the auditOnNodePage audits, then there is no need to do anymore.
+          return;
+        }
+      }
+    }
+  }
+
+  /**
+   * Log the path and user email.
+   *
+   * @param string $routeName
+   *   The route name.
+   * @param array $pageArray
+   *   The array of pages where logging is enabled.
+   * @param string $key
+   *   The key to check if in above array.
+   * @param \Drupal\node\Entity\Node $node
+   *   For type and title.
+   */
+  private function auditOnNodePage(string $routeName, array $pageArray, string $key, ?Node $node): bool {
+    // Check if the route corresponds to a node page for auditing.
+    if (!empty($pageArray) && strpos($routeName, $key) === 0 && in_array($node->getType(), $pageArray)) {
+      $this->logAuditMessage($node->getTitle());
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  /**
+   * Log the path and user email.
+   *
+   * @param string $info
+   *   The path info to include in the message.
+   */
+  private function logAuditMessage($info) {
+    $userEmail = $this->currentUser->getEmail();
+    $msg = sprintf('Potential fetch of personal data from page: %s by %s', $info, $userEmail);
+    $this->auditLogger->info('Lookup', $msg);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents(): array {
+    $events[KernelEvents::CONTROLLER][] = ['auditOnRouteChange', 1000];
+    return $events;
```

Comment:
You don't need a variable here:

```suggestion
    return [
        KernelEvents::CONTROLLER => [['auditOnRouteChange', 1000]],
    ];
```